src/viam/sdk/module/service.cpp

```diff
@@ -214,31 +212,54 @@ ::grpc::Status ModuleService_::Ready(::grpc::ServerContext* context,
     return grpc::Status();
 };
 
-ModuleService_::ModuleService_(std::string addr) {
+ModuleService::ModuleService(std::string addr) {
     module_ = std::make_shared<Module>(addr);
+    server_ = std::make_shared<Server>();
+    signal_manager_ = std::make_shared<SignalManager>();
 }
 
-void ModuleService_::start(std::shared_ptr<Server> server) {
-    const std::lock_guard<std::mutex> lock(lock_);
+ModuleService::ModuleService(int argc,
+                             char** argv,
+                             std::vector<std::shared_ptr<ModelRegistration>> registrations) {
+    if (argc < 2) {
+        throw std::runtime_error("Need socket path as command line argument");
+    }
+    module_ = std::make_shared<Module>(argv[1]);
+    server_ = std::make_shared<Server>();
+    signal_manager_ = std::make_shared<SignalManager>();
+    set_logger_severity_from_args(argc, argv);
+
+    for (const std::shared_ptr<ModelRegistration>& mr : registrations) {
+        Registry::register_model(mr);
+        add_model_from_registry(mr->api(), mr->model());
+    }
+}
+
+void ModuleService::serve() {
     const mode_t old_mask = umask(0077);
     const int sockfd = socket(AF_UNIX, SOCK_STREAM, 0);
     listen(sockfd, 10);
     umask(old_mask);
 
-    // TODO(RSDK-1742) see if we can/want to do this in an init instead
-    server->register_service(this);
+    server_->register_service(this);
     const std::string address = "unix://" + module_->addr();
-    server->add_listening_port(address);
+    server_->add_listening_port(address);
 
     module_->set_ready();
-}
+    server_->start();
+
+    BOOST_LOG_TRIVIAL(info) << "Module listening on " << module_->addr();
+    BOOST_LOG_TRIVIAL(info) << "Module handles the following API/model pairs: " << std::endl
+                            << module_->handles();
```

Comment:
These logs look like:
```
[2023-12-01 16:26:38.114981] [0x00007f78e65fdd00] [info]    Module listening on /tmp/viam-module-1887335542/MyModule.sock
[2023-12-01 16:26:38.114981] [0x00007f78e65fdd00] [info]    Module handles the following API/model pairs:
API: viam:service:summation
    Model: viam:summation:mysummation
API: viam:component:gizmo
    Model: viam:gizmo:mygizmo
API: rdk:component:base
    Model: viam:base:mybase
```
Let me know if you prefer a different format or don't like the idea of this log generally.

Reply:
They are indeed multiline and look a little gross when output by the RDK. NetCode is currently working on making module logging better, so I think I'd rather handle this at the RDK level instead of here (module authors should be able to log with `\n` and still get something vaguely greppable IMO).

Reply:
Alright, sounds good.